mbed_tools_ci_scripts/utils/third_party_licences.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Third party licences."""
+import json
+from dataclasses import dataclass
+from spdx.config import _licenses
+from license_expression import Licensing
+from typing import Iterable, cast, Optional, Iterator, List
+
+from mbed_tools_ci_scripts.utils.configuration import (
+    ConfigurationVariable,
+    configuration,
+)
+from mbed_tools_ci_scripts.utils.string_helpers import determine_similar_string_from_list
+
+
+@dataclass(order=True, frozen=True)
+class Licence:
+    """Licence descriptor.
+
+    Maps to what is defined in licenses.json in
+    https://github.com/spdx/license-list-data/blob/master/json/licenses.json
+    """
+
+    reference_number: str
+    identifier: str
+    name: str
+    is_deprecated: bool
+    is_osi_approved: bool
+    url: str
+    reference: str
+
+
+UNKNOWN_LICENCE = Licence(
+    reference_number="Unknown",
+    identifier="Unknown",
+    name="Unknown",
+    is_deprecated=True,
+    is_osi_approved=False,
+    url="Unknown",
+    reference="Unknown",
+)
+
+
+def parse_licence(licence_info: dict) -> Licence:
+    """Parses a licence entry in the Json file and translates it into a licence object."""
+    return Licence(
+        reference_number=licence_info.get("referenceNumber", ""),
+        identifier=licence_info.get("licenseId", ""),
+        name=licence_info.get("name", ""),
+        url=licence_info.get("detailsUrl", ""),
+        reference=licence_info.get("reference", ""),
+        is_deprecated=bool(licence_info.get("isDeprecatedLicenseId", False)),
+        is_osi_approved=bool(licence_info.get("isOsiApproved", True)),
+    )
+
+
+def parse_licences(licence_info: dict) -> Iterable[Licence]:
```

Comment:
It might be clearer to name this something like `iter_licenses` so it's obvious that it's a generator function.

Reply:
will do

Reply:
418326abd2b04ddfb63cea5c08fe58e7951442d4